examples/onlineboutiquemicroservice/productcatalogservice/productcatalogservice.bal

```diff
@@ -0,0 +1,120 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/grpc;
+import ballerina/io;
+
+const string CATALOG_PATH = "resources/products.json";
+
+listener grpc:Listener adListener = new (3550);
+
+@grpc:ServiceDescriptor {
+    descriptor: ROOT_DESCRIPTOR,
+    descMap: getDescriptorMap()
+}
+service "ProductCatalogService" on adListener {
+
+    remote function ListProducts() returns ListProductsResponse|error? {
+        ProductUtil utils = check new ();
+        ListProductsResponse response = {
+            products: check utils.getProducts()
+        };
+        return response;
+    }
+
+    remote function GetProduct(GetProductRequest request) returns Product|error? {
+        ProductUtil utils = check new ();
+        Product[] products = check utils.getProducts();
+        foreach Product product in products {
+            if product.id == request.id {
+                return product;
+            }
+        }
+        return {};
+    }
+
+    remote function SearchProducts(SearchProductsRequest request) returns SearchProductsResponse|error? {
+        ProductUtil utils = check new ();
```

Comment:
Rather initializing the class and loading data in every API call, shall we have it module level and read it. we follow similar approach in `routeguide` example

Reply:
Changed in https://github.com/ballerina-platform/module-ballerina-grpc/pull/392/commits/e2e6d2fc83691cacceab1f04f8fabdd186dbe4f5